sympy/testing/pytest.py

```diff
@@ -28,7 +31,6 @@
 
 if USE_PYTEST:
     raises = pytest.raises
-    warns = pytest.warns
```

Comment:
Why is this changed?

Reply:
See the updated PR description. I've updated `warns` to do things that the pytest version does not do, so I didn't think it would be a good idea to continue to allow the pytest one to be used when the tests are run in pytest. Specifically:

- `warns` checks the stacklevel to see if it is set to a level so that the warning shows the user code (can be disabled with `warns(test_stacklevel=False)`). A correctly set stacklevel helps users actually see what code of theirs is triggering the warning.
- For `SymPyDeprecationWarning`, it checks that the new `active_deprecations_target` flag actually corresponds to a real cross-reference target in the active-deprecations.md document (otherwise the URL printed in the warning message won't work).
- `warns` no longer ignores unrelated warnings. This is something that was lamented in several of the `test_pytest.py` tests, which I have updated. There was one instance of deprecated code being used in SymPy that was being silenced by this (I will open an issue about it later, but it's `plot_implicit`, and I had to XFAIL it for now).